src/main/java/mezz/jei/gui/ingredients/RecipeSlot.java

```diff
@@ -0,0 +1,312 @@
+			.ifPresent(tagEquivalent -> {
+				final TranslatableComponent acceptsAny = new TranslatableComponent("jei.tooltip.recipe.tag", tagEquivalent);
+				tooltip.add(acceptsAny.withStyle(ChatFormatting.GRAY));
+			});
+
+		return tooltip;
+	}
+
+	public void setBackground(IDrawable background) {
+		this.background = background;
+	}
+
+	public void setOverlay(IDrawable overlay) {
+		this.overlay = overlay;
+	}
+
+	public void set(List<Optional<ITypedIngredient<?>>> ingredients, List<Focus<?>> focuses) {
+		this.allIngredients = List.copyOf(ingredients);
+
+		List<ITypedIngredient<?>> matches = getMatches(focuses);
+		if (!matches.isEmpty()) {
+			this.displayIngredients = matches.stream()
+				.<Optional<ITypedIngredient<?>>>map(Optional::of)
+				.toList();
+		} else {
+			IngredientVisibility ingredientVisibility = Internal.getIngredientVisibility();
+			this.displayIngredients = this.allIngredients.stream()
+				.filter(i -> i.isEmpty() || ingredientVisibility.isIngredientVisible(i.get()))
+				.limit(MAX_DISPLAYED_INGREDIENTS)
+				.toList();
+		}
+	}
+
+	private List<ITypedIngredient<?>> getMatches(List<Focus<?>> focuses) {
+		return focuses.stream()
+			.map(this::getMatch)
+			.flatMap(Optional::stream)
+			.toList();
+	}
+
+	private <T> Optional<ITypedIngredient<?>> getMatch(Focus<T> focus) {
+		if (this.getRole() != focus.getRole()) {
+			return Optional.empty();
+		}
+		ITypedIngredient<T> focusValue = focus.getTypedValue();
+		IIngredientType<T> ingredientType = focusValue.getType();
+		List<T> typedIngredients = getIngredients(ingredientType).toList();
+		if (typedIngredients.isEmpty()) {
+			return Optional.empty();
+		}
+		IIngredientHelper<T> ingredientHelper = this.ingredientManager.getIngredientHelper(ingredientType);
+		T match = ingredientHelper.getMatch(typedIngredients, focusValue.getIngredient(), UidContext.Ingredient);
+		return TypedIngredient.create(this.ingredientManager, ingredientType, match);
+	}
+
+	public boolean isMouseOver(double recipeMouseX, double recipeMouseY) {
+		return MathUtil.contains(this.rect, recipeMouseX, recipeMouseY);
+	}
+
+	public void addTooltipCallback(IRecipeSlotTooltipCallback tooltipCallback) {
+		this.tooltipCallbacks.add(tooltipCallback);
+	}
+
+	private <T> IIngredientRenderer<T> getIngredientRenderer(IIngredientType<T> ingredientType) {
+		return Optional.of(rendererOverrides)
+			.flatMap(r -> r.getIngredientRenderer(ingredientType))
+			.orElseGet(() -> ingredientManager.getIngredientRenderer(ingredientType));
+	}
+
+	public void draw(PoseStack poseStack) {
+		cycleTimer.onDraw();
+
+		final int x = this.rect.getX();
+		final int y = this.rect.getY();
+		poseStack.pushPose();
+		{
+			poseStack.translate(x, y, 0);
+
+			if (background != null) {
+				background.draw(poseStack);
+			}
+
+			RenderSystem.enableBlend();
+
+			getDisplayedIngredient()
+				.ifPresent(ingredient -> drawIngredient(poseStack, ingredient));
+
+			if (overlay != null) {
+				overlay.draw(poseStack);
```

Comment:
Similarly to how the overlay used to work in `FluidStackRenderer`, this needs to be shifted forwards by 200 in the z direction to ensure it renders over the displayed ingredient. Also I am unsure if it is intended or not, but given at the end of `FluidStackRenderer` blend is disabled that means this overlay is being drawn with blend disabled instead of enabled similarly to how it used to be drawn when drawn as part of `FluidStackRenderer`

Reply:
This worked for me by chance, but I have added those things back to make sure it works consistently.